xbmc/pvr/PVRGUIInfo.cpp

```diff
@@ -1061,6 +1061,18 @@ bool CPVRGUIInfo::GetListItemAndPlayerBool(const CFileItem *item, const CGUIInfo
 {
   switch (info.m_info)
   {
+    case LISTITEM_HASARCHIVE:
+      if (item->IsPVRChannel())
+      {
+        bValue = item->GetPVRChannelInfoTag()->HasArchive();
+        return true;
+      }
+      else if (item->IsEPG())
+      {
+        bValue = item->GetEPGInfoTag()->IsPlayable();
```

Comment:
As an EPG tag semantically cannot have an archive, I would suggest to introduce a new guiinfo label `ListItem.IsPlayable` and to remove the EPG part from implementation of `Listitem.HasArchive`.

Reply:
Changed

Reply:
👍